PeerioTechnologies/peerio-desktop
# feat: system idle state detection

When system is idle, enable mobile push notifications.
When it becames active, disable them.

It also detects locked/unlocked screen on Windows and Mac.

#### Relevant info and issue/PR links 

https://app.clubhouse.io/peerio/story/1938/recognize-desktop-inactivity-and-send-push-notifications-once-idle 

#### Testing instructions  

1. Set system sleep/screen lock time to be at least 15 minutes so that it won't interfere with testing.
2. Launch Peerio, open dev tools.
3. Wait for 5-8 minutes while doing nothing at the computer until `The system became idle. Enabled push notifications.` log messages appear. (Can make sure here if you receive mobile notifications, but it uses the same code as when system is going to sleep.) If they won't appear, it's a bug.
4. Wiggle mouse or press a key to came out of idle state. If you didn't interact with Peerio window (e.g. no mouse over it, or keyboard inputs), the active state will be detected in about 30 seconds from the point the system becomes active. Otherwise, if you do any input in Peerio (mouse, keyboard), it should be instant.
5. Check for `The system became active (not idle). Disabled push notifications.`


----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
